Feature: View Claim Button

barebind opened the pull request:
Resolves #182

pavlovcik commented:
Currently testing with [this](https://aa123d71.pay-ubq-fi-bxp.pages.dev/?claim=W3sidHlwZSI6ImVyYzIwLXBlcm1pdCIsInBlcm1pdCI6eyJwZXJtaXR0ZWQiOnsidG9rZW4iOiIweGU5MUQxNTNFMGI0MTUxOEEyQ2U4RGQzRDc5NDRGYTg2MzQ2M2E5N2QiLCJhbW91bnQiOiIwIn0sIm5vbmNlIjoiMzQ5NjkzODQ4MDg4MTgzMjcxNzA3NjcxODY2NjQwNzYzOTIzMzg2NjY4NTg0NDUyMjc4NDA3NDE0NDEzNDA2NzU0Njc2NDMwNjI4MTMiLCJkZWFkbGluZSI6IjExNTc5MjA4OTIzNzMxNjE5NTQyMzU3MDk4NTAwODY4NzkwNzg1MzI2OTk4NDY2NTY0MDU2NDAzOTQ1NzU4NDAwNzkxMzEyOTYzOTkzNSJ9LCJ0cmFuc2ZlckRldGFpbHMiOnsidG8iOiIweDQwMDdDRTIwODNjN0YzRTE4MDk3YWVCM0EzOWJiOGVDMTQ5YTM0MWQiLCJyZXF1ZXN0ZWRBbW91bnQiOiIwIn0sIm93bmVyIjoiMHg0NENhMTVEYjEwMWZEMWMxOTQ0NjdEYjZBRjBjNjdDNkJiRjRBQjUxIiwic2lnbmF0dXJlIjoiMHhlZDU3MGU5NjMzNWE4OTVmMDhiYjk1MTNjNjhmMzQ1NGNlMmNjN2U1NDRlOTc0ZTAzYjViMzQ4YWM3ZDg4MDc4NGRkYmJhYTc0OGE1MmZhYTQ1YzBiOWQzMjc2NzEzNDQ5MTQ5ZDJhM2E2ODAxN2JhZGI2N2Q4NDM5YTk5MTVhMDFiIiwibmV0d29ya0lkIjoxMDB9XQ==).

pavlovcik commented on static/styles/rewards/claim-table.css:
```diff
@@ -134,6 +134,9 @@ table[data-claim-rendered] #controls {
   opacity: 0.5;
 }
+table[data-claim-rendered] button.hide {
+  display:none !important;
```
!important seems like a bad idea. You should use more specific selectors instead, or put this at the bottom of the css file.

barebind replied:
didn't want to make it more complicated for a simple hide function for this kind of feature, 
otherwise we need to double every selector for buttons. 
if that's okay I'm going to move the `!important` to the bottom of the css file.

pavlovcik reviewed (changes requested):
Left review.

barebind commented:
> This does not work.

we are definitely not on the same page on this one. :) 

I'm guessing it's the same issue with the one in #173 

if you are using auto generated permits via `yarn start`, then you need to insert them to database manually after they are generated, so that they can be updated later. 
this feature assumes that the permits are generated and inserted to the database by ubiquibot. 

that's the same conversation where we left on #173 

@rndquu @pavlovcik

pavlovcik commented:
> this feature assumes that the permits are generated and inserted to the database by ubiquibot.

To clarify, this is not currently implemented?

Doesn't it make more sense to write the data when we're claiming from this UI? It seems that this dependency on an external system makes it less maintainable than it needs to be.

barebind commented:
> To clarify, this is not currently implemented?
> 
> Doesn't it make more sense to write the data when we're claiming from this UI? It seems that this dependency on an external system makes it less maintainable than it needs to be.

yes inserting the permit to the db is not implemented. I really think that the permits should be inserted to the db where and when they are generated at first place which is ubiquibot. 
so that features like #172 would makes more sense and tracking permits would be more consistent. and actually I thought that it's already implemented on the ubiquibot. seems like it's not the case.

assume that a permit is generated but the beneficiary doesn't even click the link and open the UI. in that case, the permit wouldn't be listed on audit pages as unclaimed.

> It would be helpful if you can provide me the functional test URL so I can see the UI.

let me check about the functional test URL. I have no idea the automated deployments works right now. it might take some time. but in any case unless we insert the permits from ui,

pavlovcik commented:
> > To clarify, this is not currently implemented?
> > Doesn't it make more sense to write the data when we're claiming from this UI? It seems that this dependency on an external system makes it less maintainable than it needs to be.
> 
> yes inserting the permit to the db is not implemented. I really think that the permits should be inserted to the db where and when they are generated at first place which is ubiquibot. so that features like #172 would makes more sense and tracking permits would be more consistent. and actually I thought that it's already implemented on the ubiquibot. seems like it's not the case.

I'm not sure how we can confirm that your implementation works if you haven't tested it. 

> assume that a permit is generated but the beneficiary doesn't even click the link and open the UI. in that case, the permit wouldn't be listed on audit pages as unclaimed.

That makes sense. Nobody is claiming by manually passing in parameters to the etherscan write contract interface. 

> > It would be helpful if you can provide me the functional test URL so I can see the UI.
> 
> let me check about the functional test URL. I have no idea the automated deployments works right now. it might take some time.

You just need to provide the query parameter that represents the permit. We can use the latest deploy on this pull and then append the query parameter to test.

barebind commented:
> I'm not sure how we can confirm that your implementation works if you haven't tested it.

I've tested it locally as the ubiquibot is an external dependency in this case. seems like we were assuming different requirements for this task and the one before.

> You just need to provide the query parameter that represents the permit. We can use the latest deploy on this pull and then append the query parameter to test.

if it is a test on automated deployments that you want, nothing really works I guess as there is no code inserting the permit to the db. As I said before I was assuming ubiquity bot is already inserting the permits to the database. If that's not the case we first need to decide where we should be putting the db insert logic, to the ubiquibot or to the UI.

pavlovcik commented:
> I've tested it locally 

Can you explain how I can test this?

barebind commented:
> Can you explain how I can test this?

on my local

1- edit `generate-erc20-permit-url.ts` as it outputs `txData` to the console when it's run via `yarn start`
2- start ui via `yarn start`
3- insert relevant `txData` to the local supabase database
4- navigate to the UI via auto-generated 'ERC20 Testing URL' printed to the console when `yarn start`
5- test the UI

pavlovcik commented:
I gave you edit access to our Supabase why not just insert the test permit and then provide the full url here? I can only do all those steps from computer but most of the time I'm reviewing from mobile.

barebind commented:
> I gave you edit access to our Supabase why not just insert the test permit and then provide the full url here?

you can test with this [link](https://aa123d71.pay-ubq-fi-bxp.pages.dev/?claim=W3sidHlwZSI6ImVyYzIwLXBlcm1pdCIsInBlcm1pdCI6eyJwZXJtaXR0ZWQiOnsidG9rZW4iOiIweGU5MUQxNTNFMGI0MTUxOEEyQ2U4RGQzRDc5NDRGYTg2MzQ2M2E5N2QiLCJhbW91bnQiOiIwIn0sIm5vbmNlIjoiNDAzODA5Nzc2MDAzMzU3NDAxNzI2OTA4Mjk0NDExMDQwMDUxMTEwNjExMDExMTM5NDU1ODY1MTU4ODg2MzY3MTEwMjQyNzYxNjI1MzgiLCJkZWFkbGluZSI6IjExNTc5MjA4OTIzNzMxNjE5NTQyMzU3MDk4NTAwODY4NzkwNzg1MzI2OTk4NDY2NTY0MDU2NDAzOTQ1NzU4NDAwNzkxMzEyOTYzOTkzNSJ9LCJ0cmFuc2ZlckRldGFpbHMiOnsidG8iOiIweDQwMDdDRTIwODNjN0YzRTE4MDk3YWVCM0EzOWJiOGVDMTQ5YTM0MWQiLCJyZXF1ZXN0ZWRBbW91bnQiOiIwIn0sIm93bmVyIjoiMHhGNkY3QTAwZkYxQjNGRmQyZDA4NWRGZTY1NGZBQzQwZDY1ZTlERkY4Iiwic2lnbmF0dXJlIjoiMHhjNGE5NmE2ZjE3ODBkNmQ5NDNhN2U3NDIyZjRkNWU3NGIxNDQ4ODdmYTVlYzZlYzk4MzExZGExZmVhMjJjYTY5NDZjNGQwZDFmN2Y0ZDJlYjBlNWQxNGI5NGI1ZmU2ZDVlM2QyZTkwMTI3NTAwYjQyNjY4YzQ1ZTg1MzRhN2NlZDFiIiwibmV0d29ya0lkIjoxMDB9LHsidHlwZSI6ImVyYzIwLXBlcm1pdCIsInBlcm1pdCI6eyJwZXJtaXR0ZWQiOnsidG9rZW4iOiIweGU5MUQxNTNFMGI0MTUxOEEyQ2U4RGQzRDc5NDRGYTg2MzQ2M2E5N2QiLCJhbW91bnQiOiIwIn0sIm5vbmNlIjoiNDgwNzQ3MzEyMzc4MTI0MTA3NDA5MzYxMDc1NjY3MTA3NzQ5NzcwMTYzNTA5NTM0OTM1ODM1MzEyMDA1NjQ5MTU3MDkyNjgwMDExNTgiLCJkZWFkbGluZSI6IjExNTc5MjA4OTIzNzMxNjE5NTQyMzU3MDk4NTAwODY4NzkwNzg1MzI2OTk4NDY2NTY0MDU2NDAzOTQ1NzU4NDAwNzkxMzEyOTYzOTkzNSJ9LCJ0cmFuc2ZlckRldGFpbHMiOnsidG8iOiIweDQwMDdDRTIwODNjN0YzRTE4MDk3YWVCM0EzOWJiOGVDMTQ5YTM0MWQiLCJyZXF1ZXN0ZWRBbW91bnQiOiIwIn0sIm93bmVyIjoiMHhGNkY3QTAwZkYxQjNGRmQyZDA4NWRGZTY1NGZBQzQwZDY1ZTlERkY4Iiwic2lnbmF0dXJlIjoiMHgwYjYwOGU1OTdlMDc2NTQ3MzZjMmE0OTU1NjBlNTM0NTBiY2Q2OTMzMjE3NjJiZjM1NzE3YjFjNDM4MjBhZTgzNWNmMDg1ZDZiNDJjY2JlOWE2ZTE0ZjQ5ZWE4Y2FjYjc1YzZkMGEyNWU2MjhmYzYwZWE0YWZhNDI2ZjY1YzE2MDFiIiwibmV0d29ya0lkIjoxMDB9XQ==)

> Currently testing with [this](https://aa123d71.pay-ubq-fi-bxp.pages.dev/?claim=W3sidHlwZSI6ImVyYzIwLXBlcm1pdCIsInBlcm1pdCI6eyJwZXJtaXR0ZWQiOnsidG9rZW4iOiIweGU5MUQxNTNFMGI0MTUxOEEyQ2U4RGQzRDc5NDRGYTg2MzQ2M2E5N2QiLCJhbW91bnQiOiIwIn0sIm5vbmNlIjoiMzQ5NjkzODQ4MDg4MTgzMjcxNzA3NjcxODY2NjQwNzYzOTIzMzg2NjY4NTg0NDUyMjc4NDA3NDE0NDEzNDA2NzU0Njc2NDMwNjI4MTMiLCJkZWFkbGluZSI6IjExNTc5MjA4OTIzNzMxNjE5NTQyMzU3MDk4NTAwODY4NzkwNzg1MzI2OTk4NDY2NTY0MDU2NDAzOTQ1NzU4NDAwNzkxMzEyOTYzOTkzNSJ9LCJ0cmFuc2ZlckRldGFpbHMiOnsidG8iOiIweDQwMDdDRTIwODNjN0YzRTE4MDk3YWVCM0EzOWJiOGVDMTQ5YTM0MWQiLCJyZXF1ZXN0ZWRBbW91bnQiOiIwIn0sIm93bmVyIjoiMHg0NENhMTVEYjEwMWZEMWMxOTQ0NjdEYjZBRjBjNjdDNkJiRjRBQjUxIiwic2lnbmF0dXJlIjoiMHhlZDU3MGU5NjMzNWE4OTVmMDhiYjk1MTNjNjhmMzQ1NGNlMmNjN2U1NDRlOTc0ZTAzYjViMzQ4YWM3ZDg4MDc4NGRkYmJhYTc0OGE1MmZhYTQ1YzBiOWQzMjc2NzEzNDQ5MTQ5ZDJhM2E2ODAxN2JhZGI2N2Q4NDM5YTk5MTVhMDFiIiwibmV0d29ya0lkIjoxMDB9XQ==)...
> 
> This does not work.

also updated this one with placeholder `tx` on the db just in case.

pavlovcik commented:
> > I gave you edit access to our Supabase why not just insert the test permit and then provide the full url here?
> 
> you can test with this [link](https://aa123d71.pay-ubq-fi-bxp.pages.dev/?claim=W3sidHlwZSI6ImVyYzIwLXBlcm1pdCIsInBlcm1pdCI6eyJwZXJtaXR0ZWQiOnsidG9rZW4iOiIweGU5MUQxNTNFMGI0MTUxOEEyQ2U4RGQzRDc5NDRGYTg2MzQ2M2E5N2QiLCJhbW91bnQiOiIwIn0sIm5vbmNlIjoiNDAzODA5Nzc2MDAzMzU3NDAxNzI2OTA4Mjk0NDExMDQwMDUxMTEwNjExMDExMTM5NDU1ODY1MTU4ODg2MzY3MTEwMjQyNzYxNjI1MzgiLCJkZWFkbGluZSI6IjExNTc5MjA4OTIzNzMxNjE5NTQyMzU3MDk4NTAwODY4NzkwNzg1MzI2OTk4NDY2NTY0MDU2NDAzOTQ1NzU4NDAwNzkxMzEyOTYzOTkzNSJ9LCJ0cmFuc2ZlckRldGFpbHMiOnsidG8iOiIweDQwMDdDRTIwODNjN0YzRTE4MDk3YWVCM0EzOWJiOGVDMTQ5YTM0MWQiLCJyZXF1ZXN0ZWRBbW91bnQiOiIwIn0sIm93bmVyIjoiMHhGNkY3QTAwZkYxQjNGRmQyZDA4NWRGZTY1NGZBQzQwZDY1ZTlERkY4Iiwic2lnbmF0dXJlIjoiMHhjNGE5NmE2ZjE3ODBkNmQ5NDNhN2U3NDIyZjRkNWU3NGIxNDQ4ODdmYTVlYzZlYzk4MzExZGExZmVhMjJjYTY5NDZjNGQwZDFmN2Y0ZDJlYjBlNWQxNGI5NGI1ZmU2ZDVlM2QyZTkwMTI3NTAwYjQyNjY4YzQ1ZTg1MzRhN2NlZDFiIiwibmV0d29ya0lkIjoxMDB9LHsidHlwZSI6ImVyYzIwLXBlcm1pdCIsInBlcm1pdCI6eyJwZXJtaXR0ZWQiOnsidG9rZW4iOiIweGU5MUQxNTNFMGI0MTUxOEEyQ2U4RGQzRDc5NDRGYTg2MzQ2M2E5N2QiLCJhbW91bnQiOiIwIn0sIm5vbmNlIjoiNDgwNzQ3MzEyMzc4MTI0MTA3NDA5MzYxMDc1NjY3MTA3NzQ5NzcwMTYzNTA5NTM0OTM1ODM1MzEyMDA1NjQ5MTU3MDkyNjgwMDExNTgiLCJkZWFkbGluZSI6IjExNTc5MjA4OTIzNzMxNjE5NTQyMzU3MDk4NTAwODY4NzkwNzg1MzI2OTk4NDY2NTY0MDU2NDAzOTQ1NzU4NDAwNzkxMzEyOTYzOTkzNSJ9LCJ0cmFuc2ZlckRldGFpbHMiOnsidG8iOiIweDQwMDdDRTIwODNjN0YzRTE4MDk3YWVCM0EzOWJiOGVDMTQ5YTM0MWQiLCJyZXF1ZXN0ZWRBbW91bnQiOiIwIn0sIm93bmVyIjoiMHhGNkY3QTAwZkYxQjNGRmQyZDA4NWRGZTY1NGZBQzQwZDY1ZTlERkY4Iiwic2lnbmF0dXJlIjoiMHgwYjYwOGU1OTdlMDc2NTQ3MzZjMmE0OTU1NjBlNTM0NTBiY2Q2OTMzMjE3NjJiZjM1NzE3YjFjNDM4MjBhZTgzNWNmMDg1ZDZiNDJjY2JlOWE2ZTE0ZjQ5ZWE4Y2FjYjc1YzZkMGEyNWU2MjhmYzYwZWE0YWZhNDI2ZjY1YzE2MDFiIiwibmV0d29ya0lkIjoxMDB9XQ==)
> 
> > Currently testing with [this](https://aa123d71.pay-ubq-fi-bxp.pages.dev/?claim=W3sidHlwZSI6ImVyYzIwLXBlcm1pdCIsInBlcm1pdCI6eyJwZXJtaXR0ZWQiOnsidG9rZW4iOiIweGU5MUQxNTNFMGI0MTUxOEEyQ2U4RGQzRDc5NDRGYTg2MzQ2M2E5N2QiLCJhbW91bnQiOiIwIn0sIm5vbmNlIjoiMzQ5NjkzODQ4MDg4MTgzMjcxNzA3NjcxODY2NjQwNzYzOTIzMzg2NjY4NTg0NDUyMjc4NDA3NDE0NDEzNDA2NzU0Njc2NDMwNjI4MTMiLCJkZWFkbGluZSI6IjExNTc5MjA4OTIzNzMxNjE5NTQyMzU3MDk4NTAwODY4NzkwNzg1MzI2OTk4NDY2NTY0MDU2NDAzOTQ1NzU4NDAwNzkxMzEyOTYzOTkzNSJ9LCJ0cmFuc2ZlckRldGFpbHMiOnsidG8iOiIweDQwMDdDRTIwODNjN0YzRTE4MDk3YWVCM0EzOWJiOGVDMTQ5YTM0MWQiLCJyZXF1ZXN0ZWRBbW91bnQiOiIwIn0sIm93bmVyIjoiMHg0NENhMTVEYjEwMWZEMWMxOTQ0NjdEYjZBRjBjNjdDNkJiRjRBQjUxIiwic2lnbmF0dXJlIjoiMHhlZDU3MGU5NjMzNWE4OTVmMDhiYjk1MTNjNjhmMzQ1NGNlMmNjN2U1NDRlOTc0ZTAzYjViMzQ4YWM3ZDg4MDc4NGRkYmJhYTc0OGE1MmZhYTQ1YzBiOWQzMjc2NzEzNDQ5MTQ5ZDJhM2E2ODAxN2JhZGI2N2Q4NDM5YTk5MTVhMDFiIiwibmV0d29ya0lkIjoxMDB9XQ==)...
> > This does not work.
> 
> also updated this one with placeholder `tx` on the db just in case.

Am I supposed to claim it? It has two rewards embedded, bot are claimable according to the UI.

barebind commented:
> Am I supposed to claim it? It has two rewards embedded, bot are claimable according to the UI.

in order to test whole process I supposed you would want to claim it. I extracted your address from previous link. so if you want to claim you can. if you only want to test ui i also updated the previous link that you said didn't work. that's the supposed version of ui when you claim the permit.

pavlovcik commented:
Love to see it. 

<img width="834" alt="image" src="https://github.com/ubiquity/pay.ubq.fi/assets/4975670/813c5126-2995-485c-9e37-20b0bae5612a">

> rewards are generated via script with the amount 0. by default it creates two permits so i didn't edit it.

This didn't happen for me so I don't understand why that would be the case for you.
